main.cpp

```diff
@@ -1632,6 +1683,18 @@ static int main_inner(int argc, char *argv[])
 	cbs.add("--msl-pad-fragment-output", [&args](CLIParser &) { args.msl_pad_fragment_output = true; });
 	cbs.add("--msl-domain-lower-left", [&args](CLIParser &) { args.msl_domain_lower_left = true; });
 	cbs.add("--msl-argument-buffers", [&args](CLIParser &) { args.msl_argument_buffers = true; });
+	cbs.add("--msl-argument-buffer-pad",
+	        [&args](CLIParser &parser)
+	        {
+		        MSLResourceBinding rez_bind;
+		        rez_bind.stage = stage_to_execution_model(parser.next_string());
+		        rez_bind.basetype = str_to_spir_base_type(parser.next_string());
+		        rez_bind.desc_set = parser.next_uint();
+		        rez_bind.binding = parser.next_uint();
+		        rez_bind.count = parser.next_uint();
+		        args.msl_resource_bindings.push_back(std::move(rez_bind));
```

Comment:
This looks a bit weird, since this is declaring argument buffer resource bindings, not the padding option. Based on this code, I would expect it to be called --msl-argument-buffer-binding and a separate --msl-argument-buffer-pad option that enables the padding thing as a single bool.

Reply:
Either way, the PR is still marked draft, so I'll resume when it's marked ready.

Reply:
Is it valid to specify padding and not also specify resources, or to specify resources for another reason?

In this change the padding option is enabled if resources are specified, line 1250.

The test logic I added to `test_shaders.py` does not exercise the intended `if (!resource.descriptor_alias)` branch, so I have some learning to do before being confident in this codebase.

Reply:
Hm, ok. Either way, I didn't expect the complexity of adding this test to be this great, so I'm okay with just taking the fix as-is, without the test. You can leave out the test commit. I'll have to think about how to expose this later.